lxd/apparmor.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+
+	"github.com/lxc/lxd/shared"
+
+	log "gopkg.in/inconshreveable/log15.v2"
+)
+
+const (
+	APPARMOR_CMD_LOAD   = "r"
+	APPARMOR_CMD_UNLOAD = "R"
+)
+
+var aaPath = shared.VarPath("security", "apparmor")
+
+const DEFAULT_POLICY = `
+#include <tunables/global>
+profile lxd-%s flags=(attach_disconnected,mediate_deleted) {
+    #include <abstractions/lxc/container-base>
+
+    # user input raw.apparmor below here
+    %s
+}`
+
+func AAProfileName(c *containerLXD) string {
+	return fmt.Sprintf("lxd-%s", c.name)
+}
+
+// getProfileContent generates the apparmor profile template from the given
+// container. This includes the stock lxc includes as well as stuff from
+// raw.apparmor.
+func getProfileContent(c *containerLXD) string {
+	rawApparmor, ok := c.config["raw.apparmor"]
+	if !ok {
+		rawApparmor = ""
+	}
+
+	return fmt.Sprintf(DEFAULT_POLICY, c.name, rawApparmor)
+}
+
+func runApparmor(command string, profile string) error {
+	cmd := exec.Command("apparmor_parser", []string{
```

Comment:
That check should ideally happen at daemon startup time and print a log entry if apparmor support is disabled.